runtime/vm/ContinuationHelpers.cpp

```diff
@@ -44,16 +44,22 @@ createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 	J9VMContinuation *continuation = NULL;
 
 	/* First check if local cache is available. */
-	if (NULL != currentThread->cachedContinuation) {
-		continuation = currentThread->cachedContinuation;
-		currentThread->cachedContinuation = NULL;
-	} else {
+	if (NULL != currentThread->continuationT1Cache) {
+		for (U_32 i = 0; i < vm->continuationT1Size; i++) {
+			if (NULL != currentThread->continuationT1Cache[i]) {
+				continuation = currentThread->continuationT1Cache[i];
+				currentThread->continuationT1Cache[i] = NULL;
+				break;
+			}
+		}
+	}
+	if (NULL == continuation) {
 		/* Greedily try to use first available cache from global array. */
-		for (U_32 i = 0; i < vm->continuationArraySize; i++) {
-			continuation = vm->globalContinuationCacheArray[i];
+		for (U_32 i = 0; i < vm->continuationT2Size; i++) {
```

Comment:
Is there a rough range for `continuationT1Size` or `continuationT2Size`? If it is a large number for linear searching, did we consider adding a counter to the next available cache?

Reply:
T2 caching is a shared non-blocking cache, all access is done via CAS which doesn't allow for a next counter as that break the assumption on atomic updates (in the case if frequent access) as you will have to update 2 fields for every store. And an extra search when the current counter is consumed during cache look up.

Reply:
T1 size should be kept relatively small, while T2 size depends on the application. This is just a prototype work to evaluate the benefit of different cache size. the assumption is mostly user should be using the default sizes set by VM.